### PR TITLE
Bump the github.com/onsi/ginkgo dependency to v2

### LINKED ISSUE #997: Bump github.com/onsi/ginkgo to v2
Update the "github.com/onsi/ginkgo" dependency from the v1 to v2 version. The OLM runtime repository has already done this migration in https://github.com/operator-framework/operator-lifecycle-manager/pull/2732.

### PR DESCRIPTION
**Pull request by timflannagan**
**Description of the change:**
Update the repository from ginkgo v1 to ginkgo v2.

**Motivation for the change:**
Align ginkgo version with the other major OLM repositories.

**Reviewer Checklist**
- [x] Implementation matches the proposed design, or proposal is updated to match implementation
- [x] Sufficient unit test coverage 
- [x] Sufficient end-to-end test coverage
- [x] Docs updated or added to `/docs` 
- [x] Commit messages sensible and descriptive

Closes #997

### CODE REVIEW

**Comment by timflannagan** [comment]
It looks like that codecov check is off given https://app.codecov.io/gh/operator-framework/operator-registry/commit/f8f3f89000b8341095723f677e65b2528295f6ec reports that the difference is with some veneer semver code, but there are zero changes to anything that touches any of the pkg/test files in that domain.

File: docs/contributors/e2e_tests.md [hunk 1]
```diff
@@ -127,11 +127,11 @@ TIP: use a non-dynamic `kind` server by using `kind get kubeconfig --name "kind"
    KUBECONFIG="$HOME/.kube/config" DOCKER_REGISTRY_HOST=localhost:443 make build e2e TEST='builds and manipulates bundle and index images'
    ```
 
-1. If you want a quick way to ensure that your TEST regex argument will work, you can bypass the 
+1. If you want a quick way to ensure that your TEST regex argument will work, you can bypass the
 make file and use `-dryRun` with `-focus` and see if the regex would trigger your specific test(s).
 
    ```bash
-   GOFLAGS="-mod=vendor" go run github.com/onsi/ginkgo/ginkgo --v --randomizeAllSpecs --randomizeSuites --race -dryRun -focus 'builds and manipulates bundle and index images' -tags=json1 ./test/e2e
+   GOFLAGS="-mod=vendor" go run github.com/onsi/ginkgo/v2/ginkgo --v --randomizeAllSpecs --randomizeSuites --race -dryRun -focus 'builds and manipulates bundle and index images' -tags=json1 ./test/e2e
```
Comment (ankitathomas) [inline docs/contributors/e2e_tests.md:134]: This still uses the deprecated flag names

**Review by ankitathomas — APPROVED** [review]
Looks good, just one small nit about the docs

File: docs/contributors/e2e_tests.md (hunk 1 above)
Comment (timflannagan) [reply docs/contributors/e2e_tests.md]: Yup, nice catch. Updated.

**Pull request merged**